tests/testInitialSetupConsent.php

```diff
@@ -0,0 +1,291 @@
+<?php
+/**
+ * Contains Class TestInitialSetupConsent.
+ *
+ * @package WP-Auth0
+ *
+ * @since 3.8.1
+ */
+
+use PHPUnit\Framework\TestCase;
+
+/**
+ * Class TestInitialSetupConsent.
+ */
+class TestInitialSetupConsent extends TestCase {
+
+	use httpHelpers {
+		httpMock as protected httpMockDefault;
+	}
+
+	use RedirectHelpers;
+
+	use SetUpTestDb;
+
+	/**
+	 * Instance of WP_Auth0_Options.
+	 *
+	 * @var WP_Auth0_Options
+	 */
+	public static $opts;
+
+	/**
+	 * WP_Auth0_ErrorLog instance.
+	 *
+	 * @var WP_Auth0_ErrorLog
+	 */
+	protected static $error_log;
+
+	/**
+	 * Setup for entire test class.
+	 */
+	public static function setUpBeforeClass() {
+		parent::setUpBeforeClass();
+		self::$opts      = WP_Auth0_Options::Instance();
+		self::$error_log = new WP_Auth0_ErrorLog();
+	}
+
+	/**
+	 * Test that an invalid state is redirected to the right place.
+	 */
+	public function testThatInvalidStateRedirectsProperly() {
+		$this->startRedirectHalting();
+
+		$setup_consent = new WP_Auth0_InitialSetup_Consent( self::$opts );
+		$test_domain   = 'test-wp.auth0.com';
+		$test_token    = implode( '.', [ uniqid(), uniqid(), uniqid() ] );
+
+		$caught_redirect = [];
+		try {
+			$setup_consent->callback_with_token( $test_domain, $test_token, 'invalid_state' );
```

Comment:
what's the previous context? how can you ensure that `invalid_state` is an invalid state? normally, you'd manually set that state before this call.

Reply:
There are only two values that `callback_with_token` accept here ... not actually state but that's the name used in the method.